### PR TITLE
5713 - Datagrid Row Selection Arrow Keys

### PR DESCRIPTION
**Pull request by tjamesallen15**
**Explain the _details_ for making this change. What existing problem does the pull request solve?**

This PR will add datagrid feature using arrow keys to select.

**Related github/jira issue (required)**:
Closes https://github.com/infor-design/enterprise/issues/5713

**Steps necessary to review your pull request (required)**:
- Pull this branch, build, and run the app
- Go to http://localhost:4000/components/datagrid/example-key-row-select.html
- Click one row
- Use Arrow Keys (Up and Down) to move selection

**Included in this Pull Request**:
- [ ] An e2e or functional test for the bug or feature.
- [x] A note to the change log.

### CODE REVIEW

File: src/components/notification-badge/notification-badge.js [hunk 1]
```diff
@@ -23,7 +23,6 @@ const NOTIFICATION_BADGE_DEFAULTS = {
  * @param {string} [settings] The settings element.
  * @param {string} [settings.position] The placement of notification badge.
  * @param {string} [settings.color] The color of the notification badge.
- * @param {string} [settings.icon] The icon to display.
```
Comment (tmcconechy) [inline src/components/notification-badge/notification-badge.js:26]: Did you mean to remove?

File: app/views/components/datagrid/example-key-row-select.html [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        columns: columns,
+        dataset: data,
+        selectable: 'single',
+        cellNavigation: false,
+        disableRowDeselection: true,
+        clickToSelect: true,
+				keyRowSelect: true,
+        toolbar: {title: 'Data Grid Header Title', results: true, personalize: true, actions: true, rowHeight: true, keywordFilter: true,  collapsibleFilter: true}
+      }).on('selected', function (e, args) {
+        if (args[0]) {
```
Comment (tmcconechy) [inline app/views/components/datagrid/example-key-row-select.html:47]: Can we remove the args[0] so the console log happens in the test?

**Review by tmcconechy — CHANGES_REQUESTED** [review]
A couple small things. In addition can you check the events i am not sure if this is 1000% an issue but

- tab into the grid with a row selected -> fires selected
- tab out also fires selected (but this is hard to test with the args[0]) on it

File: src/components/notification-badge/notification-badge.js (hunk 1 above)
Comment (tjamesallen15) [reply src/components/notification-badge/notification-badge.js:26]: Yep. I forgot to remove this on my last change on notification badge. I just happen to notice it while coding. Should I make another PR for this one? My bad.

**Comment by tjamesallen15** [comment]
> Also maybe a test for this one?

will do tim!

File: src/components/notification-badge/notification-badge.js (hunk 1 above)
Comment (tmcconechy) [reply src/components/notification-badge/notification-badge.js:26]: No its fine - just wanted to check.

**Review by tmcconechy — APPROVED** [review]
The only thing i see is..

- tab in and arrow down a few rows
- tab out of the grid
- tab back in - event fires when already selected

But i think lets let the app team try this because i think changing that could cause issues so lets leave it. So just making a note 👍🏻

**Review by tmcconechy — APPROVED** [review]

**Review by ericangeles — APPROVED** [review]

**Review by ericangeles — APPROVED** [review]

**Pull request merged**